src/Azure.Functions.Cli/Helpers/ExtensionsHelper.cs

```diff
@@ -63,7 +63,12 @@ public static IEnumerable<ExtensionPackage> GetExtensionPackages()
                 }
             }
 
-            packages.Add("ExtensionsMetadataGeneratorPackage", Constants.ExtensionsMetadataGeneratorPackage);
+            // We only need extensionsMetadatGeneratorPackage, if there is a trigger type that requires an extension
```

Comment:
Nit: not just trigger type, there are extensions that have only input or output bindings.